### PR TITLE
Deleted ".. ago" btns, using moment api for date

### PR DESCRIPTION
**Pull request by amabelleS**
# Description - delete buttons in Realtime Map Page #62 
Deleted "5/10 minutes ago" buttons.

Code that is no longer in use on comment now. may need another commit after approved changes.
Such as the formatTime() function and its uses. state variables "from" & "to" are defined to 5 minutes ago, and now. using moment api instead of the Date object.
MinuteSelector - setting from & to onChange to keep the difference. Starting (first render) at 5 minutes difference.

## screenshots
image

### CODE REVIEW

File: src/pages/RealtimeMapPage.tsx [hunk 1]
```diff
@@ -52,10 +51,10 @@ export const colorIcon = ({ operator_id, name }: { operator_id: string; name?: s
...
 }
 
-function formatTime(time: string | number | Date) {
-  const date = new Date(time).toISOString()
-  return date
-}
+// function formatTime(time: string | number | Date) {
+//   const date = new Date(time).toISOString() // takes as 3 hours back...
+//   return date
+// }
```
Comment (NoamGaash) [inline src/pages/RealtimeMapPage.tsx:57]: it it's not used, don't keep it as a comment

File: src/pages/RealtimeMapPage.tsx [hunk 2]
```diff
@@ -73,14 +72,19 @@ export default function RealtimeMapPage() {
   }
 
   //TODO (another PR and another issue) load from url like in another pages.
-  const [from, setFrom] = useState('2023-05-01T12:00:00+02:00') // arbitrary default value. this date is not important
-  const [to, setTo] = useState('2023-05-01T12:01:00+02:00')
+  const fiveMinutesAgo = moment().subtract(5, 'minutes')
+  const now = moment()
```
Comment (NoamGaash) [inline src/pages/RealtimeMapPage.tsx:76]: define these constants outside the component, on top.
we don't want a new date to be created every render

**Review by NoamGaash — CHANGES_REQUESTED** [review]
looks amazing!
please remove all comments

**Comment by amabelleS** [comment]
Done:) I also changed line 187+188 to use moment api

image

File: src/pages/RealtimeMapPage.tsx [hunk 3]
```diff
@@ -43,20 +42,15 @@ export const colorIcon = ({ operator_id, name }: { operator_id: string; name?: s
...
     html: `
     <div class="bus-icon-container">
-      <div class="bus-icon-circle">
-        <img src="${path}" alt="${name}" />
-      </div>
-      <div class="operator-name">${name}</div>
+    <div class="bus-icon-circle">
+    <img src="${path}" alt="${name}" />
+    </div>
+    <div class="operator-name">${name}</div>
```
Comment (NoamGaash) [inline src/pages/RealtimeMapPage.tsx:48]: please restore the indentation

File: src/pages/RealtimeMapPage.tsx [hunk 4]
```diff
@@ -138,21 +132,21 @@ export default function RealtimeMapPage() {
         </Grid>
         <Grid xs={5}>
           <DateSelector
-            timeValid={moment(from.slice(0, 16))} // remove timezone and seconds
+            timeValid={to} // 
             setTimeValid={(ts) => {
-              const value = ts ? ts.format() : ''
-              setFrom(value)
-              setTo(formatTime(+new Date(value) + (+new Date(to) - +new Date(from)))) // keep the same time difference
+              // const value = ts ? ts.format() : ''
```
Comment (NoamGaash) [inline src/pages/RealtimeMapPage.tsx:137]: please remove code in comments

File: src/pages/RealtimeMapPage.tsx [hunk 5]
```diff
@@ -162,39 +156,25 @@ export default function RealtimeMapPage() {
         </Grid>
         <Grid xs={6}>
           <MinuteSelector
-            num={(+new Date(to) - +new Date(from)) / 1000 / 60}
-            setNum={(num) => setTo(formatTime(+new Date(from) + +num * 1000 * 60))}
+            num={to.diff(from)/1000/60}
+            setNum={(num) => {
+              setFrom(moment().subtract(num, 'minutes'))
+              setTo(moment())
+            }}
```
Comment (NoamGaash) [inline src/pages/RealtimeMapPage.tsx:163]: nice, thaks!

File: src/pages/RealtimeMapPage.tsx [hunk 6]
```diff
@@ -162,39 +156,25 @@ export default function RealtimeMapPage() {
...
           />
         </Grid>
         <Grid xs={1}>
           <Label text={TEXTS.minutes} />
         </Grid>
         {/* Buttons */}
         {/*TODO (another PR another issue)
-          1) discussion if the buttons need to be 5 minutes from now or from the 'from' state.
+          1) discussion if the buttons need to be 5 minutes from now or from the 'from' state. 
+          => Done.
```
Comment (NoamGaash) [inline src/pages/RealtimeMapPage.tsx:172]: if it's done - just delete it :)

File: src/pages/RealtimeMapPage.tsx [hunk 7]
```diff
@@ -162,39 +156,25 @@ export default function RealtimeMapPage() {
...
           2) recommended use MomentJs API instead Date API. 'moment().subtract(5, 'minutes')'.
-          3) use text `TEXTS`.
+          => Done.
+          3) use text `TEXTS`. => ?? were ?
```
Comment (NoamGaash) [inline src/pages/RealtimeMapPage.tsx:175]: see this:
https://github.com/hasadna/open-bus-map-search/pull/95
we can do it on another PR

File: src/pages/RealtimeMapPage.tsx [hunk 8]
```diff
@@ -204,8 +184,8 @@ export default function RealtimeMapPage() {
             {loaded} {`- `}
             {TEXTS.show_x_bus_locations} {` `}
             {TEXTS.from_time_x_to_time_y
-              .replace('XXX', new Date(from).toLocaleTimeString())
-              .replace('YYY', new Date(to).toLocaleTimeString())}
+              .replace('XXX', moment(from).format('hh:mm A')) // cleaner code, moment api
```
Comment (NoamGaash) [inline src/pages/RealtimeMapPage.tsx:187]: remove the comment - it might confuse future contributors

**Review by NoamGaash — APPROVED** [review]
great job!
approved with comments

**Comment by NoamGaash** [comment]
to see lint errors - run `yarn run lint` :)

File: src/pages/RealtimeMapPage.tsx (hunk 3 above)
Comment (amabelleS) [reply src/pages/RealtimeMapPage.tsx:48]: image

File: src/pages/RealtimeMapPage.tsx (hunk 4 above)
Comment (amabelleS) [reply src/pages/RealtimeMapPage.tsx:137]: We need another PR/issue for the DateSelector & TimeSelector: Now when you set the time to another date or hour, it does not set the state variables to & from accordingly. We need to set "to" to the timeSelected by the user, and from to the number of minutes difference (5 min default, or what the user selected with MinuteSelector component).

Comment (NoamGaash) [reply src/pages/RealtimeMapPage.tsx:137]: sounds like it should be solved in this PR, to avoid regressions
also, adding tests to this behaviour would be very beneficial

Comment (amabelleS) [reply src/pages/RealtimeMapPage.tsx:137]: As I was testing it today I realized that the date, hours, and minutes difference selectors were not in sync. I thought we needed another PR to fix it, but I think it's okay now. I made a few changes to make it work. 
Now, when the user picks another date, then the minutes gap he selects will be for the same date.

**Comment by amabelleS** [comment]
> to see lint errors - run `yarn run lint` :)

image

**Comment by NoamGaash** [comment]
first option - google "how to lint on save", and add your IDE name to the search query.
other option - change the number of spaces where you get the errors.
foe rxample, change `/1000/` to ` / 1000 / `

**Comment by NoamGaash** [comment]
great job!
thank you.
